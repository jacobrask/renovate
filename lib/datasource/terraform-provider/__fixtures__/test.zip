version https://git-lfs.github.com/spec/v1
oid sha256:b7871907b3f8222cda1cfb48930d96e89d82adf965da3e934036488be64a475b
size 618
