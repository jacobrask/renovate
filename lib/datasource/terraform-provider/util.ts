export const RepositoryRegex = /^(?:(?<namespace>[^\s]+)\/)?(?<dependency>[^\s]+)$/;
