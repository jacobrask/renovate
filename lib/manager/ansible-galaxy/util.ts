export const newBlockRegEx = /^\s*-\s*((\w+):\s*(.*))$/;
export const blockLineRegEx = /^\s*((\w+):\s*(.*))$/;
export const galaxyDepRegex = /.+\..+/;
