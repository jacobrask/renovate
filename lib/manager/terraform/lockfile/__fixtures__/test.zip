version https://git-lfs.github.com/spec/v1
oid sha256:1721d74eaf36c245148394532dc93854e5568352efad27e57361864e7edfded7
size 473638
