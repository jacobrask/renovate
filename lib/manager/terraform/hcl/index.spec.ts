import { loadFixture } from '../../../../test/util';
import { parseHCL, parseJSON } from './index';

const modulesTF = loadFixture('modules.tf');
const resourcesTF = loadFixture('resources.tf');
const resourcesTFJSON = loadFixture('resources.tf.json');
const lockedVersion = loadFixture('lockedVersion.tf');

describe('manager/terraform/hcl/index', () => {
  describe('parseHCL()', () => {
    it('should return flat modules', async () => {
      const res = await parseHCL(modulesTF);
      expect(Object.keys(res.module)).toBeArrayOfSize(6);
      expect(res).toMatchSnapshot();
    });

    it('should return nested terraform block', async () => {
      const res = await parseHCL(lockedVersion);
      expect(res).toMatchSnapshot({
        terraform: [
          {
            required_providers: [
              {
                aws: {},
                azurerm: {},
                kubernetes: {},
              },
            ],
          },
        ],
      });
    });

    it('should return resource blocks', async () => {
      const res = await parseHCL(resourcesTF);
      expect(res).toMatchSnapshot({
        resource: {
          docker_container: {
            foo: {},
            invalid: {},
          },
          docker_service: {
            foo: [
              {
                name: 'foo-service',
                task_spec: [
                  {
                    container_spec: {},
                  },
                ],
                endpoint_spec: [
                  {
                    ports: {},
                  },
                ],
              },
            ],
          },
        },
      });
    });
  });

  describe('parseJSON', () => {
    it('should parse json', async () => {
      const res = await parseJSON(resourcesTFJSON);
      expect(res).toMatchSnapshot({
        resource: {
          aws_instance: {
            example: {},
          },
        },
      });
    });
  });
});
