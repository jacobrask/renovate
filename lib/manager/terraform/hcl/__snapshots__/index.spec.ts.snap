// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/terraform/hcl/index parseHCL() should return flat modules 1`] = `
Object {
  "module": Object {
    "bar": Array [
      Object {
        "source": "github.com/hashicorp/example?ref=next",
      },
    ],
    "consul": Array [
      Object {
        "source": "hashicorp/consul/aws",
        "version": "0.1.0",
      },
    ],
    "foo": Array [
      Object {
        "source": "github.com/hashicorp/example?ref=v1.0.0",
      },
    ],
    "repo-with-dot": Array [
      Object {
        "source": "github.com/hashicorp/example.2.3?ref=v1.0.0",
      },
    ],
    "repo-with-dot-and-git-suffix": Array [
      Object {
        "source": "github.com/hashicorp/example.2.3.git?ref=v1.0.0",
      },
    ],
    "repo-with-non-semver-ref": Array [
      Object {
        "source": "github.com/githubuser/myrepo//terraform/modules/moduleone?ref=tfmodule_one-v0.0.9",
      },
    ],
  },
}
`;

exports[`manager/terraform/hcl/index parseHCL() should return nested terraform block 1`] = `
Object {
  "terraform": Array [
    Object {
      "required_providers": Array [
        Object {
          "aws": Object {
            "source": "aws",
            "version": "~> 3.0",
          },
          "azurerm": Object {
            "version": "~> 2.50.0",
          },
          "kubernetes": Object {
            "source": "terraform.example.com/example/kubernetes",
            "version": ">= 1.0",
          },
        },
      ],
    },
  ],
}
`;

exports[`manager/terraform/hcl/index parseHCL() should return resource blocks 1`] = `
Object {
  "resource": Object {
    "docker_container": Object {
      "foo": Array [
        Object {
          "image": "nginx:1.7.8",
          "name": "foo",
        },
      ],
      "invalid": Array [
        Object {
          "name": "foo",
        },
      ],
    },
    "docker_service": Object {
      "foo": Array [
        Object {
          "endpoint_spec": Array [
            Object {
              "ports": Array [
                Object {
                  "target_port": "8080",
                },
              ],
            },
          ],
          "name": "foo-service",
          "task_spec": Array [
            Object {
              "container_spec": Array [
                Object {
                  "image": "repo.mycompany.com:8080/foo-service:v1",
                },
              ],
            },
          ],
        },
      ],
    },
  },
}
`;

exports[`manager/terraform/hcl/index parseJSON should parse json 1`] = `
Object {
  "resource": Object {
    "aws_instance": Object {
      "example": Object {
        "provisioner": Array [
          Object {
            "local-exec": Object {
              "command": "echo 'Hello World' >example.txt",
            },
          },
          Object {
            "file": Object {
              "destination": "/tmp/example.txt",
              "source": "example.txt",
            },
          },
          Object {
            "remote-exec": Object {
              "inline": Array [
                "sudo install-something -f /tmp/example.txt",
              ],
            },
          },
        ],
      },
    },
  },
}
`;
